sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruningSuite.scala

```diff
@@ -87,6 +87,27 @@ class PruningSuite extends HiveComparisonTest with BeforeAndAfter {
     Seq("key"),
     Seq.empty)
 
+  createPruningTest("Column pruning - explode with aggregate",
+    "SELECT name, sum(d) AS sumd FROM person LATERAL VIEW explode(data) d AS d GROUP BY name",
+    Seq("name", "sumd"),
+    Seq("name","data"),
+    Seq.empty)
```

Comment:
I don't think we need to go all the way to hive to create an integration test.  Instead create some unit tests in `...catalyst.optimizer.`